### ISSUE TITLE
OCIO Maker Clips to Working Space

### ISSUE THREAD

**Issue by raveninthesky** [EatTheFuture/image_tools#18]
OCIO Maker currently clips images when the input gamut is different from the working space gamut. This makes OCIO configs from OCIO Maker risky for live action projects, such as VFX with more than one camera, or when using a working space other than the camera. 

Clipping the gamut at this early stage will likely damage footage outside of the working space bounds. This becomes an issue when footage needs to round trip back to log for color grading after vfx compositing.

**Comment by thatcherfreeman** [comment]
Imo clipping is definitely a risk and likely should not be unnecessarily introduced by the OCIO config. Many camera primaries extend outside the common working spaces with the assumption that you'll handle these super-saturated colors later in the pipeline. Clipping at the input will then introduce unrecoverable distortions and potentially give compositors the wrong idea of what backplate they're trying to match.

**Comment by cessen** [comment]
You're right, I should add an option to make the transforms not do gamut clipping/compression.

In my defense, the assumed use case for OCIO Maker was creating configurations for 3d rendering software, where negative RGB values (such as those produced by out-of-gamut colors) can be neither preserved nor meaningfully handled, but are very likely to cause rendering artifacts.  And for that reason I'll still be leaving clipping/compression in as an option.

But if the configurations are used in compositing software and e.g. only relatively simple operations are performed on the values (such gain, alpha-over, etc.) the indeed those out-of-gamut colors can be meaningfully handled and preserved.  And in such cases you are of course 100% right that you don't want OCIO clipping that color data.

Thank you!

**Comment by thatcherfreeman** [comment]
Thank you, having it as an option I think would make everyone happy. Maybe there could be a tooltip too to advise when enabling clipping would or would not be preferable.

**Comment by raveninthesky** [comment]
Thank you! Adding that option would be very helpful in workflows where footage is composited and sent to color for final image formation.
